crates/egui/src/widgets/plot/mod.rs

```diff
@@ -651,10 +737,148 @@ impl Plot {
+        let mut x_axis_widgets = Vec::<XAxisWidget>::new();
+        let mut y_axis_widgets = Vec::<YAxisWidget>::new();
+        {
+            // determine absolute rectangle for each axis label widget
+            // widget cnt per border of plot in order left, top, right, bottom
+            struct WidgetCnt {
+                left: usize,
+                top: usize,
+                right: usize,
+                bottom: usize,
+            }
+            let mut widget_cnt = WidgetCnt {
+                left: 0,
+                top: 0,
+                right: 0,
+                bottom: 0,
+            };
+            if show_axes.x {
+                for cfg in &x_axes {
+                    let size_y = Vec2::new(0.0, cfg.thickness());
+                    let rect = match cfg.placement {
+                        axis::Placement::Default => {
+                            let off = widget_cnt.bottom as f32;
+                            widget_cnt.bottom += 1;
+                            Rect {
+                                min: plot_rect.left_bottom() + size_y * off,
+                                max: plot_rect.right_bottom() + size_y * (off + 1.0),
+                            }
+                        }
+                        axis::Placement::Opposite => {
+                            let off = widget_cnt.top as f32;
+                            widget_cnt.top += 1;
+                            Rect {
+                                min: plot_rect.left_top() - size_y * (off + 1.0),
+                                max: plot_rect.right_top() - size_y * off,
+                            }
+                        }
+                    };
+                    x_axis_widgets.push(XAxisWidget::new(cfg.clone(), rect));
+                }
+            }
+            if show_axes.y {
+                for cfg in &y_axes {
+                    let size_x = Vec2::new(cfg.thickness(), 0.0);
+                    let rect = match cfg.placement {
+                        axis::Placement::Default => {
+                            let off = widget_cnt.left as f32;
+                            widget_cnt.left += 1;
+                            Rect {
+                                min: plot_rect.left_top() - size_x * (off + 1.0),
+                                max: plot_rect.left_bottom() - size_x * off,
+                            }
+                        }
+                        axis::Placement::Opposite => {
+                            let off = widget_cnt.right as f32;
+                            widget_cnt.right += 1;
+                            Rect {
+                                min: plot_rect.right_top() + size_x * off,
+                                max: plot_rect.right_bottom() + size_x * (off + 1.0),
+                            }
+                        }
+                    };
+                    y_axis_widgets.push(YAxisWidget::new(cfg.clone(), rect));
+                }
+            }
+        }
```

Comment:
The `show_dyn` functions is already way too long. Can you try to move out some of this code into helper function so as not to increase the complexity of `show_dyn` further?